doc/userguide/rules/frame-keywords.rst

```diff
@@ -0,0 +1,172 @@
+**************
+Frame Keywords
+**************
+
+Overview
+========
+
+Frame support is introduced with Suricata 7.0. `Frame` is a generic term that can represent any unit of network data one is interested in. Frames work as "stream annotations", meaning that the engine is able to know what type of record exists at a specific offset in the stream.
+
+Consider a rule like::
+
+    alert smb ... flow:to_server; frame:smb1.data; content:"some smb 1 issue";
+
+This rule will only inspect the data portion of SMB1 frames. It will not affect
+any other protocol, and it won't need special patterns to "search" for the
+SMB1 frame in the raw stream.
+
+This means that one can write rules to match on more specialized types, making rule writing less tedious and enhancing rules performance - this is especially true if a ruleset has all rules that could use frames converted to this new keyword.
+
+Frame keywords will work for application layer protocols that have frame support added.
+As such, the list of available frame keywords vary in a per-protocol basis.
+
+Currently, protocols that support frame keywords are:
+
+- `http`_
+- `sip`_
+- `smb`_
+- `telnet`_
+- `tls`_
+
+We have work in progress for supporting ``dcerpc``, ``dns``, ``nfs`` and ``pgsql``, and for the Suricata 7.0 release we want to offer frame support for all parsers.
+
+Examples:
+
+- tls.pdu
+- smb.smb2.hdr
+- smb.smb3.data
+
+Naming conventions:
+
+- hdr: header (only present in case of successfully parsed records)
+- data: payload (only present in case of successfully parsed records)
+- pdu: the whole record, for the given app-layer protocol. Always created, even if we have a malformed record.
+
+Frame rule keywords act as sticky buffers and as such they can be combined with any of the content modifier keywords, such as many of the described for `payload <https://suricata.readthedocs.io/en/latest/rules/payload-keywords.html?#payload-keywords>`_.
```

Comment:
Add a section explaining with what keywords frame keywords are not expected to work.

Reply:
examples:
```
[ERRCODE: SC_ERR_INVALID_SIGNATURE(39)] - can't mix app-layer buffer and frame inspection
[ERRCODE: SC_ERR_INVALID_SIGNATURE(39)] - can't mix app-layer buffer and frame inspection
```